### PR TITLE
Continuous demo delivery

### PR DESCRIPTION
**Pull request by safchain**

### CODE REVIEW

**Comment by safchain** [comment]
run devstack-plugin

**Comment by safchain** [comment]
run devstack-plugin

**Comment by lebauce** [comment]
run-devstack

**Comment by lebauce** [comment]
run devstack

**Comment by safchain** [comment]
rerun selenium-tests

**Comment by safchain** [comment]
rerun tests

**Comment by safchain** [comment]
run cdd-overview

**Comment by safchain** [comment]
run cdd-overview

**Comment by safchain** [comment]
run cdd-overview

**Comment by safchain** [comment]
run cdd-overview

**Comment by safchain** [comment]
rerun selenium-tests

**Comment by safchain** [comment]
rerun selenium-tests

**Comment by safchain** [comment]
rerun selenium-tests

**Comment by safchain** [comment]
rerun cdd-overview

**Comment by safchain** [comment]
run cdd-overview

**Comment by safchain** [comment]
run cdd-overview

**Comment by safchain** [comment]
run cdd-overview

**Comment by safchain** [comment]
rerun tests

**Comment by safchain** [comment]
run cdd-overview

**Comment by safchain** [comment]
run cdd-overview

**Comment by safchain** [comment]
run cdd-overview

File: statics/js/fake-mouse.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  this.moveOn(el, function() {
+    self.click(callback);
+  });
+};
+
+FakeMousePointer.prototype.moveOn = function(el, callback) {
+	var self = this;
+
+  var br = el.getBoundingClientRect();
+	this.moveTo((br.left + br.right) / 2, (br.top + br.bottom) / 2, 2, function() {
```
Comment (lebauce) [inline statics/js/fake-mouse.js]: nit: indentation

File: tests/cdd_test.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+		return
+	}
+
+	if err = delaySec(1, sh.enableFakeMousePointer()); err != nil {
+		t.Error(err)
+		return
+	}
+
+	// start recording
+	sh.startVideoRecod("overview")
```
Comment (lebauce) [inline tests/cdd_test.go]: nit: s/Recod/Record

File: tests/tests.go [hunk 3]
```diff
@@ -362,6 +367,627 @@ func ping(t *testing.T, context *TestContext, ipVersion int, src string, dst str
...
+	s.webdriver.Quit()
+
+	tearDownCmds := []helper.Cmd{
+		{"docker stop grid", true},
+		{"docker rm -f grid", true},
+	}
+	helper.ExecCmds(s.t, tearDownCmds...)
+}
+
+//elgalu/seleniu
```
Comment (lebauce) [inline tests/tests.go]: nit: remove line

File: statics/js/fake-mouse.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  x += 1; y += 1;
+
+  var path = document.createElementNS('http://www.w3.org/2000/svg','path');
+
+  var x1 = this.x, y1 = this.y, x2 = x, y2 = y;
+  var xx = x2 - x1, yy = y2 - y1;
+  var l = Math.sqrt(xx * xx + yy * yy);
+  var cx = (x1 + x2) / 2, cy  = (y1 + y2) / 2;
+  var angle = Math.atan2(yy, xx);
+	// curve height
```
Comment (lebauce) [inline statics/js/fake-mouse.js]: nit: fix indentation

File: statics/js/fake-mouse.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  path.setAttribute('d','M ' + x1 + ' ' + y1 + 'Q '+ cpx + ' ' + cpy + ' ' + x2 + ' ' + y2);
+  var len = path.getTotalLength();
+
+  var move = function(percent, step) {
+    var pt = path.getPointAtLength(percent / 100 * len);
+    self.pointer.css({ left: pt.x, top: pt.y });
+
+    percent += step;
+    if (percent <= 100) {
+			var ms = self.easeInOutQuart(percent/100);
```
Comment (lebauce) [inline statics/js/fake-mouse.js]: fix identation

**Comment by safchain** [comment]
run cdd-overview

**Comment by safchain** [comment]
run cdd-overview

**Pull request merged**